agent/engine/engine_unix_integ_test.go

```diff
@@ -1908,3 +1915,107 @@ func TestHostResourceManagerLaunchTypeBehavior(t *testing.T) {
+// This is because the VerifyManifestPulledStateChange helper methods wait for events on the task engine's state channels,
+// but on Windows, these are skipped due to no local registry setup, so the test progresses to its assertions faster than intended.
+func TestTaskResourceDependencyOnRestart(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	taskArn := "arn:aws:ecs:us-east-1:123456789012:task/testTaskResourceDependencyOnRestart"
+	testCredentialsID := "test-execution-credentials-id"
+	secretValueFrom := "arn:aws:ssm:us-east-1:123456789012:parameter/test-secret"
+	secretValue := "mock-secret-value"
+
+	// Create a task with a SSM secret resource dependency
+	testTask := CreateTestTask(taskArn)
+	testTask.Containers[0].Image = "busybox:latest" // Override container image
+	testTask.Containers[0].Secrets = []apicontainer.Secret{
+		{
+			Name:      "test-secret",
+			ValueFrom: secretValueFrom,
+			Provider:  "ssm",
+			Region:    "us-east-1",
+		},
+	}
+	testTask.Containers[0].TransitionDependenciesMap = make(map[apicontainerstatus.ContainerStatus]apicontainer.TransitionDependencySet)
+	testTask.ResourcesMapUnsafe = make(map[string][]taskresource.TaskResource)
+	testTask.SetExecutionRoleCredentialsID(testCredentialsID)
+
+	// Start engine with default config
+	taskEngine, done, _, credentialsManager := setupWithDefaultConfig(t)
+	defer done()
+
+	// Set up SSM mocks
+	ssmClientCreator := mockssmfactory.NewMockSSMClientCreator(ctrl)
+	mockSSMClient := mockssm.NewMockSSMClient(ctrl)
+	mockCreds := &credentials.TaskIAMRoleCredentials{
+		ARN: taskArn,
+		IAMRoleCredentials: credentials.IAMRoleCredentials{
+			RoleArn:       "arn:aws:iam::123456789012:role/execution-role",
+			CredentialsID: testCredentialsID,
+		},
+	}
+	ssmOutput := &ssm.GetParametersOutput{
+		Parameters: []ssmtypes.Parameter{
+			{
+				Name:  aws.String(secretValueFrom),
+				Value: aws.String(secretValue),
+			},
+		},
+	}
+	ssmClientCreator.EXPECT().NewSSMClient("us-east-1", mockCreds.IAMRoleCredentials, ipcompatibility.NewIPv4OnlyCompatibility()).Return(mockSSMClient, nil).AnyTimes()
+	mockSSMClient.EXPECT().GetParameters(gomock.Any(), gomock.Any()).Do(func(ctx context.Context, in *ssm.GetParametersInput, optFns ...func(*ssm.Options)) {
+		require.Equal(t, []string{secretValueFrom}, in.Names)
+	}).Return(ssmOutput, nil).AnyTimes()
+
+	// Configure engine with SSM client creator
+	taskEngine.(*DockerTaskEngine).resourceFields = &taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator:   ssmClientCreator,
+			CredentialsManager: credentialsManager,
+		},
+	}
+
+	// Add task to engine without credentials
+	go taskEngine.AddTask(testTask)
+
+	// Verify task progresses but gets blocked waiting for execution credentials
+	VerifyContainerManifestPulledStateChange(t, taskEngine)
+	VerifyTaskManifestPulledStateChange(t, taskEngine)
```

Comment:
Do we have an assertion for the task getting blocked waiting on exec role creds?